lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js

```diff
@@ -228,6 +229,39 @@ class UnusedBytes extends Audit {
     };
   }
 
+  /**
+   * Utility function to estimate transfer size and cache calculation.
+   * @param {Map<string, number>} transferRatioByUrl
+   * @param {string} url
+   * @param {LH.Artifacts} artifacts
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {LH.Crdp.Network.ResourceType=} resourceType
+   */
+  static async estimateTransferRatio(transferRatioByUrl, url, artifacts,
```

Comment:
My previous attempt of mapping a `resource` to `transfer` size for `wastedBytesByUrl` was lacking: completely overlooked the wasted bytes values in the `items` themselves (it was resource size; only `wastedBytesByUrl` map was transfer size).

I took a step back and realized we just need a nicer way to estimate the transfer ratio.

@patrickhulce sry for the back/forth on this one

Reply:
ah nice catch! +1 to sharing this piece between the two

I will say though that this has an unusual API compared to its surroundings to live as a utility like this (cache state that you keep outside of this method but this staticish method also updates it?). also seems overly specific to scripts and not stylesheet content, HTML, etc?

If we want to land this quickly I think I'd prefer to just leave it on duplicated-javascript and have legacy-javascript.js pull it in so other audits don't get the idea that it's generic and ready to roll. a name like `estimateTransferRatioForScript` might help too 👍 

how do you feel about that?

Reply:
also kinda weird that duplicated-javascript only really cares about the transfer size ultimately, so maybe a future refactor could augment that helper with some script ability instead of this new one?

Reply:
> also seems overly specific to scripts and not stylesheet content, HTML, etc?

Doh, hehe passing `resourceType` is silly given this.

> how do you feel about that?

yeah sg.